lib/DataFetcher.dart

```diff
@@ -73,8 +76,46 @@ class DataFetcher {
     return new List();
   }
 
-  static Future<bool> uploadGrades(String filePath) {
-    return Future.value(true); // TODO send to server
+  static fetchFormData(FormClass.FormData data) async {
+    Dio dio = new Dio();
+
+    final String serverEndPoint = _api + "/user/profile";
```

Comment:
Remove the leading "/" as the _api variable has it already, which is causing the request to fail with a `404 Not found`.